src/components/Amass/Amass.tsx

```diff
@@ -0,0 +1,172 @@
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+            setPid(""); // Clear the child process pid reference.
+            setLoading(false); // Cancel the loading overlay.
+            setAllowSave(true); // Allow Saving as the output is finalised.
+            setHasSaved(false);
+        },
+        [handleProcessData]
+    );
+
+    /**
+     * Function to handle completion of saving output to a text file.
+     * Sets the hasSaved flag to true and disallows further saving.
+     */
+    const handleSaveComplete = () => {
+        setHasSaved(true);
+        setAllowSave(false);
+    };
+
+    /**
+     * onSubmit: Asynchronous handler for the form submission event.
+     * It sets up and triggers the Amass tool with the given parameters.
+     * Once the command is executed, the results or errors are displayed in the output.
+     *
+     * @param {FormValuesType} values - The form values, containing the domain to scan.
+     */
+    const onSubmit = (values: FormValuesType) => {
+        setAllowSave(false);
+        setLoading(true);
+        const args = ["enum", "-d", values.domain];
+        CommandHelper.runCommandGetPidAndOutput("amass", args, handleProcessData, handleProcessTermination)
+            .then(({ pid, output }) => {
+                setPid(pid);
+                setOutput(output);
+            })
+            .catch((error) => {
+                setLoading(false);
+                setOutput(`Error: ${error.message}`);
+            });
+    };
+
+    /**
+     * Function to clear output and reset save status.
+     */
+    const clearOutput = useCallback(() => {
+        setOutput("");
+        setHasSaved(false);
+        setAllowSave(false);
+    }, [setOutput]);
+
+    return (
+        <RenderComponent
+            title={title}
+            description={description}
+            steps={steps}
+            tutorial={tutorial}
+            sourceLink={sourceLink}
+        >
+            {!loadingModal && (
+                <InstallationModal
+                    isOpen={opened}
+                    setOpened={setOpened}
+                    feature_description={description}
+                    dependencies={dependencies}
+                ></InstallationModal>
+            )}
+            <form onSubmit={form.onSubmit((values) => onSubmit(values))}>
+                {LoadingOverlayAndCancelButton(loading, pid)}
+                <Stack>
+                    <TextInput label="Enter the domain to scan" required {...form.getInputProps("domain")} />
+                    {SaveOutputToTextFile_v2(output, allowSave, hasSaved, handleSaveComplete)}
+                    <Button type="submit">Run Amass</Button>
```

Comment:
The convention is "Start <title of tool>"
```suggestion
                    <Button type="submit">Start {title}</Button>
```